### PR TITLE
new proposal for short paths in Win

### PR DESCRIPTION
**Pull request by memsharded**
NOT to merge yet.

This would be a new proposal for the damn path length limit in Win. It should help with #379, #375, #374, #371.

It doesnt have tests, and has some magic values, as the path to store things. Using UUIDs at the moment, but maybe something better can be used.

It should allow removing the other approach, it is not related at all, nor uses any part of it.

Just for feedback on the approach.

### CODE REVIEW

**Comment by lasote** [comment]
The idea looks good, continue with it ;)

**Comment by memsharded** [comment]
Now the proposal is improved, ready to be merged if ok.

File: conans/tools.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def unzip(filename, destination="."):
                 # the limit is 230 to account for 10 chars of SHA in b/xxxx (in shorted paths)
                 # and 20 chars for build system extra subfolders (cmake mytarget.dir/debug, etc)
                 # Win limit is 260
-                if len(file_.filename) + len(full_path) > 230:
+                if len(file_.filename) + len(full_path) >= 260:
```
Comment (memsharded) [inline conans/tools.py]: This unzip is now able to unzip larger paths, but then it warns in the copy from source => build, that also has a check for path length. I have tested to build Boost in Win (unzipping boost had this issue) and works fine. Maybe the copy source=>build is slow, but nothing really changed there, it was already slow I guess.

**Comment by memsharded** [comment]
This is also a solution for #311, and will help for Qt in Win, avoiding consumers having to do any configuration.

File: conans/client/proxy.py [hunk 2]
```diff
@@ -88,11 +88,8 @@ def _refresh():
 
         # check if it is in disk
         conanfile_path = self._paths.conanfile(conan_reference)
-        is_min_path = conan_reference in self._paths.short_path_refs
```
Comment (lasote) [inline conans/client/proxy.py]: Finally you remove the old short_paths?

File: conans/tools.py [hunk 3]
```diff
@@ -64,7 +64,7 @@ def unzip(filename, destination="."):
                 # the limit is 230 to account for 10 chars of SHA in b/xxxx (in shorted paths)
                 # and 20 chars for build system extra subfolders (cmake mytarget.dir/debug, etc)
                 # Win limit is 260
```
Comment (lasote) [inline conans/tools.py]: So, remove or modify the # comments

**Comment by lasote** [comment]
Just take a look to the "#comments" comment that I did. It works well and looks good.

File: conans/client/proxy.py (hunk 2 above)
Comment (memsharded) [inline conans/client/proxy.py]: Yep, the old short_paths is totally removed.

**Pull request merged**